Add Seaformer model

inderpreetsingh01 opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes #21668 
Seaformer is a two-branch architecture with Squeeze enhanced Axial Transformer for semantic segmentation on mobile devices.
<br>
Supersedes #21774 


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? #21668 
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?
@alaradirik thanks for offering help with this PR, please let me know about any changes required.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker and @younesbelkada
- vision models: @amyeroberts
- speech models: @sanchit-gandhi
- graph models: @clefourrier

Library:

- flax: @sanchit-gandhi
- generate: @gante
- pipelines: @Narsil
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @sgugger

Integrations:

- deepspeed: HF Trainer: @stas00, Accelerate: @pacman100
- ray/raytune: @richardliaw, @amogkam

Documentation: @sgugger, @stevhliu and @MKhalusova

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @sanchit-gandhi
- PyTorch: @sgugger
- TensorFlow: @Rocketknight1

 -->

inderpreetsingh01 committed:
Add model

inderpreetsingh01 committed:
doc and tests

inderpreetsingh01 committed:
doc and tests

alaradirik commented:
Hi @inderpreetsingh01, thank you! You can ping me once the PR is ready is to be reviewed. 

You can follow the [official guidelines](https://huggingface.co/docs/transformers/add_new_model) to learn how to prepare the configuration, image processor and modeling files to replicate the original work such that forward propagating an image through the HF and original implementation yields the same results.

alaradirik commented:
> # What does this PR do?
> Fixes #21668 Seaformer is a two-branch architecture with Squeeze enhanced Axial Transformer for semantic segmentation on mobile devices. Supersedes #21774
> 
> ## Before submitting
> * [ ]  This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
> * [x]  Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
>   Pull Request section?
> * [x]  Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? [Add SeaFormer model #21668](https://github.com/huggingface/transformers/issues/21668)
> * [x]  Did you make sure to update the documentation with your changes? Here are the
>   [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
>   [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
> * [ ]  Did you write any new necessary tests?
> 
> ## Who can review?
> @alaradirik thanks for offering help with this PR, please let me know about any changes required.

The PR is just initialized using SegFormer, I can do a review once the SeaFormer model is implemented.

inderpreetsingh01 committed:
Merge remote-tracking branch 'upstream/main' into add_seaformer_model

inderpreetsingh01 committed:
updated modeling, configuration and conversion script

inderpreetsingh01 committed:
updated modeling, configuration and conversion script

inderpreetsingh01 commented:
Hi @alaradirik, I have added seaformer implementation in modeling file and updated the conversion and configuration scripts, I have ran a forward pass in notebook and output is same as the original seaformer model. Can you please review it and let me know of any changes required? I am yet to do the testing part.

alaradirik commented on src/transformers/models/seaformer/configuration_seaformer.py:
```diff
@@ -0,0 +1,192 @@
+        depths=[3, 3, 3],
+        num_labels = 150,
+        channels = [32, 64, 128, 192, 256, 320],
+        cfgs = [
```
Some of the attributes are not in the docstrings above, could you update it and include all attributes (cfgs, is_dw)?

We also make the attribute names as descriptive as possible for ease of use (e.g. is_dw -> is_depthwise)

alaradirik commented on src/transformers/models/seaformer/configuration_seaformer.py:
```diff
@@ -0,0 +1,192 @@
+        self.semantic_loss_ignore_index = semantic_loss_ignore_index
+        self.reshape_last_stage = kwargs.get("reshape_last_stage", True)
+        
+class SeaformerOnnxConfig(OnnxConfig):
+    torch_onnx_minimum_version = version.parse("1.11")
+
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("pixel_values", {0: "batch", 1: "num_channels", 2: "height", 3: "width"}),
+            ]
+        )
+
+    @property
+    def atol_for_validation(self) -> float:
+        return 1e-4
+
+    @property
+    def default_onnx_opset(self) -> int:
+        return 12
```
No need for the Onnx class anymore :)
```suggestion
```

alaradirik commented on src/transformers/models/seaformer/convert_seaformer_original_to_pytorch.py:
```diff
@@ -0,0 +1,208 @@
+
+from transformers import (
+    SeaformerConfig,
+    SeaformerFeatureExtractor,
```
We are replacing all `XXXFeatureExtractor` classes with `XXXImageProcessor`
```suggestion
    SeaformerImageProcessor,
```

alaradirik commented on src/transformers/models/seaformer/convert_seaformer_original_to_pytorch.py:
```diff
@@ -0,0 +1,208 @@
+        raise ValueError(f"Size {size} not supported")
+
+    # load feature extractor (only resize + normalize)
+    feature_extractor = SeaformerFeatureExtractor(
```
```suggestion
    feature_extractor = SeaformerImageProcessor(
```

alaradirik commented on src/transformers/models/seaformer/convert_seaformer_original_to_pytorch.py:
```diff
@@ -0,0 +1,208 @@
+
+    # verify logits
+    assert logits.shape == expected_shape
+    print(logits[0, :3, :3, :3])
```
```suggestion
```

alaradirik commented on src/transformers/models/seaformer/convert_seaformer_original_to_pytorch.py:
```diff
@@ -0,0 +1,208 @@
+    Path(pytorch_dump_folder_path).mkdir(exist_ok=True)
+    model.save_pretrained(pytorch_dump_folder_path)
+    feature_extractor.save_pretrained(pytorch_dump_folder_path)
+
```
Could you also add a push_to_hub argument to upload the model to the hub?

It's really convenient as it takes care of repo initialization too:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/maskformer/convert_maskformer_swin_to_pytorch.py#L302

alaradirik commented on src/transformers/models/seaformer/convert_seaformer_original_to_pytorch.py:
```diff
@@ -0,0 +1,208 @@
+    parser.add_argument(
+        "--pytorch_dump_folder_path", default=None, type=str, help="Path to the folder to output PyTorch model."
+    )
+    args = parser.parse_args()
```
```suggestion
    parser.add_argument(
        "--push_to_hub", action="store_true", help="Whether or not to push the converted model to the 🤗 hub."
    )
    args = parser.parse_args()
```

alaradirik commented on src/transformers/utils/dummy_vision_objects.py:
```diff
@@ -401,6 +401,20 @@ def __init__(self, *args, **kwargs):
         requires_backends(self, ["vision"])
 
 
+class SeaformerFeatureExtractor(metaclass=DummyObject):
+    _backends = ["vision"]
+
+    def __init__(self, *args, **kwargs):
+        requires_backends(self, ["vision"])
+
+
```
No need to include this class anymore:
```suggestion
```

alaradirik commented on src/transformers/utils/dummy_pt_objects.py:
```diff
@@ -5783,6 +5783,51 @@ def load_tf_weights_in_roformer(*args, **kwargs):
+SEAFORMER_PRETRAINED_MODEL_ARCHIVE_LIST = None
+
+
+class SeaformerDecodeHead(metaclass=DummyObject):
+    _backends = ["torch"]
+
+    def __init__(self, *args, **kwargs):
+        requires_backends(self, ["torch"])
+
+
```
Same here, no need to export utility classes:
```suggestion
```

alaradirik commented on src/transformers/models/seaformer/feature_extraction_seaformer.py:
```diff
@@ -0,0 +1,33 @@
+# coding=utf-8
```
Please see my previous comments, we are deprecating the FeatureExtractor classes for vision models, you can delete this file

alaradirik commented on src/transformers/models/seaformer/image_processing_seaformer.py:
```diff
@@ -0,0 +1,486 @@
+        do_reduce_labels: bool = False,
+        **kwargs,
+    ) -> None:
+        if "reduce_labels" in kwargs:
+            warnings.warn(
+                "The `reduce_labels` parameter is deprecated and will be removed in a future version. Please use "
+                "`do_reduce_labels` instead.",
+                FutureWarning,
+            )
+            do_reduce_labels = kwargs.pop("reduce_labels")
```
No need to include this for a new model, we include such warnings to existing models to ensure backward compatibility:
```suggestion
```

alaradirik commented on src/transformers/models/seaformer/image_processing_seaformer.py:
```diff
@@ -0,0 +1,486 @@
+        self.image_std = image_std if image_std is not None else IMAGENET_DEFAULT_STD
+        self.do_reduce_labels = do_reduce_labels
+
+    @property
+    def reduce_labels(self):
+        warnings.warn(
+            "The `reduce_labels` property is deprecated and will be removed in a v4.27. Please use "
+            "`do_reduce_labels` instead.",
+            FutureWarning,
+        )
+        return self.do_reduce_labels
```
Same here, we can get rid of backward compatibility related parts:
```suggestion
```

alaradirik commented on src/transformers/models/seaformer/image_processing_seaformer.py:
```diff
@@ -0,0 +1,486 @@
+        )
+        return self.do_reduce_labels
+
+    @classmethod
+    def from_dict(cls, image_processor_dict: Dict[str, Any], **kwargs):
+        """
+        Overrides the `from_dict` method from the base class to make sure `do_reduce_labels` is updated if image
+        processor is created using from_dict and kwargs e.g. `SeaformerImageProcessor.from_pretrained(checkpoint,
+        reduce_labels=True)`
+        """
+        image_processor_dict = image_processor_dict.copy()
+        if "reduce_labels" in kwargs:
+            image_processor_dict["reduce_labels"] = kwargs.pop("reduce_labels")
+        return super().from_dict(image_processor_dict, **kwargs)
```
Same here, this function is for backward compatibility in existing models and can be removed:
```suggestion
```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+    It ensures that all layers have a channel number that is divisible by 8
+    It can be seen here:
+    https://github.com/tensorflow/models/blob/master/research/slim/nets/mobilenet/mobilenet.py
+    :param v:
+    :param divisor:
+    :param min_value:
+    :return:
```
```suggestion
```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+    # See all Seaformer models at https://huggingface.co/models?filter=seaformer
+]
+
+def _make_divisible(v, divisor, min_value=None):
```
Could you replace `v` with a more descriptive variable name?
```suggestion
def _make_divisible(value, divisor, min_value=None):
```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        
+    def forward(self, x):
+        return self.bn(self.c(x))
+
```
Needs additional spacing, you can automatically correct for these with:
```
make style
make quality
make repo-consistency
```

```suggestion


```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        nn.init.constant_(bn.bias, 0)
+        self.add_module('bn', bn)
+        
+    def forward(self, x):
```
We use more descriptive argument names (usually pixel_values for the first layer/block and hidden_states for others) within the forward functions and use type casting.
```suggestion
    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+    
+
+class SeaformerConv2d_BN(nn.Module):
+    def __init__(self, a, b, ks=1, stride=1, pad=0, dilation=1,
```
More descriptive argument and variable names please:
in_dim instead of a, out_dim instead of b, etc.

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        self.stride = stride
+        self.dilation = dilation
+        self.groups = groups
+        self.add_module('c', nn.Conv2d(
+            a, b, ks, stride, pad, dilation, groups, bias=bias))
+        
+        bn = nn.BatchNorm2d(b)
```
Same here, layer names should be more descriptive (convolution instead of c, batchnorm instead of bn)

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+            a, b, ks, stride, pad, dilation, groups, bias=bias))
+        
+        bn = nn.BatchNorm2d(b)
+        nn.init.constant_(bn.weight, bn_weight_init)
+        nn.init.constant_(bn.bias, 0)
+        self.add_module('bn', bn)
```
We use XXXPretrainedModel class to initialize the parameters optimally, please refer to other modeling files to see examples.

```suggestion
```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        return "p={}".format(self.drop_prob)
+    
+
+class SeaformerConv2d_BN(nn.Module):
```
SeaformerConv2D or SeaFormerEmbedding would be better class names

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        self.fc2 = SeaformerConv2d_BN(hidden_features, out_features)
+        self.drop = nn.Dropout(drop)
+
+    def forward(self, x):
```
Please see my comment above, use more descriptive argument names and type casting please:
```suggestion
    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        self.expand_ratio = expand_ratio
+        assert stride in [1, 2]
+
+        if activations is None:
+            activations = nn.ReLU
```
Could this be accessed via the configuration by passing config as an argument? (config.activation)

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+
+        layers = []
+        if expand_ratio != 1:
+            # pw
```
Could be expanded to be more descriptive

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+                activation()
+            )
+            
+        self.cfgs = cfgs
```
We generally use the config to access configuration parameters wherever possible, so class initialization arguments such as cfgs and activation should be eliminated ideally

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        self.pos_embed = nn.Parameter(torch.randn([1, dim, shape]))
+
+    def forward(self, x):
+        B, C, N = x.shape
```
```suggestion
        batch_size, num_channels, dim = hidden_states.shape
```

alaradirik commented on src/transformers/models/seaformer/modeling_seaformer.py:
```diff
@@ -0,0 +1,764 @@
+        
+        return x
+    
+class h_sigmoid(nn.Module):
+    def __init__(self, inplace=True):
+        super(h_sigmoid, self).__init__()
+        self.relu = nn.ReLU6(inplace=inplace)
+
+    def forward(self, x):
+        return self.relu(x + 3) / 6
```
Is this activation function specific to SeaFormer? In either case, you can add this class to the transformers activations file and import it:
https://github.com/huggingface/transformers/blob/main/src/transformers/activations.py

See an example over here:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/maskformer/modeling_maskformer.py#L551

alaradirik reviewed (commented):
Thank you for adding this model!

I think the PR is in good shape overall but there are some issues that needs to be addressed. To summarize the main points:
- SeaformerForImageClassification is not implemented and this is causing modeling test errors. Could you implement this class or remove all references to it?
- We use self-descriptive variable and layer names, as well as type casting for all model classes and functions

Once you're done, you can run the following commands to ensure your PR passes all CI tests:
```
make style
make quality
make repo-consistency

pytest tests/models/seaformer/test_image_processing_seaformer.py
RUN_SLOW=True pytest tests/models/seaformer/test_modeling_seaformer.py
```

I can also double check the model output and the post-processing results if you upload a converted model to the hub and let me know.

Thanks again :)

inderpreetsingh01 commented:
Hi @alaradirik thanks for the detailed review :) I have uploaded the converted model to the hub here Inderpreet01/seaformer-semantic-segmentation-large, will work on your comments and update the pr.
Thanks

alaradirik commented:
> Hi @alaradirik thanks for the detailed review :) I have uploaded the converted model to the hub here Inderpreet01/seaformer-semantic-segmentation-large, will work on your comments and update the pr. Thanks

Thank you! Feel free to ping me when you'd like me to do the final review

inderpreetsingh01 committed:
updated readme files

inderpreetsingh01 committed:
updated readme files

inderpreetsingh01 committed:
Merge remote-tracking branch 'upstream/main' into add_seaformer_model

inderpreetsingh01 committed:
Merge remote-tracking branch 'upstream/main' into add_seaformer_model

inderpreetsingh01 committed:
updated docs and __init__ file

inderpreetsingh01 committed:
Merge remote-tracking branch 'upstream/main' into add_seaformer_model

inderpreetsingh01 committed:
updated configuration removed feature_extraction file

inderpreetsingh01 committed:
updated configuration removed feature_extraction file

inderpreetsingh01 committed:
removed feature extraction references and added push_to_hub

inderpreetsingh01 committed:
updated image_processing_seaformer to remove backward compatibility parts

inderpreetsingh01 committed:
Update src/transformers/models/seaformer/modeling_seaformer.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
test

inderpreetsingh01 committed:
Merge branch 'add_seaformer_model' of https://github.com/inderpreetsingh01/transformers into add_seaformer_model

inderpreetsingh01 committed:
added type casting

inderpreetsingh01 committed:
updated layer names, added function to activations and other changes

inderpreetsingh01 committed:
code formatting and tests

inderpreetsingh01 committed:
changes

inderpreetsingh01 committed:
changes

inderpreetsingh01 committed:
remove ImageClassification references

inderpreetsingh01 committed:
removed SeformerLayer references

inderpreetsingh01 committed:
removed seaformerfeature extractor reference

inderpreetsingh01 committed:
removed image classification from seaformer.mdx file

inderpreetsingh01 committed:
updated docs

inderpreetsingh01 commented:
Hi @alaradirik I have worked on the changes you mentioned, two tests are failing in test_modeling_seaformer.py

SeaformerModelTest::test_initialization - AssertionError: -6.169999778649071e-06 not found in [0.0, 1.0]
I have normally initialized the parameters so negative values are expected.

SeaformerModelTest::test_config - ValueError: The following keys were not properly set in the config:
label2id and id2label are having 150 items but it is expecting 1 item in test_configuration_common.py [config_common_kwargs](https://github.com/huggingface/transformers/blob/c612628045822f909020f7eb6784c79700813eda/tests/test_configuration_common.py#L78-L79) dictionary is having id2label and label2id key dictionary with one item as value.

Can you please help me with them thanks.

inderpreetsingh01 committed:
code quality

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_21819). All of your documentation changes will be reflected on that endpoint.

inderpreetsingh01 commented:
Also I have worked on the checks and most of them are successful, will need your help with the remaining three checks. thanks.

inderpreetsingh01 committed:
udated semantic segmentation docs

inderpreetsingh01 committed:
code quality

alaradirik commented:
Hi @inderpreetsingh01, I took a look at the code and failed tests and saw that some of the failures are due to unrelated models. Could you rebase to main by clicking on the _Synch fork_ button on your [branch](https://github.com/inderpreetsingh01/transformers/tree/add_seaformer_model)?

The modeling test failure stemming from the label mapping is probably just due to setting a `num_labels` attribute within `SeaformerConfig`. All config classes inherit from the `PretrainedConfig` class, which computes the `num_labels` based on the `id2label` and `label2id` attributes, which are initialized to have 2 labels by default. You should remove the `num_labels` attribute and overwrite the default `id2label` and `label2id` attributes within the conversion script. You can take a look at the configuration, conversion and test scripts of MaskFormer and Mask2Former to see how that's done.

Hope this helps!

inderpreetsingh01 committed:
Merge branch 'huggingface:main' into add_seaformer_model

inderpreetsingh01 committed:
test changes

inderpreetsingh01 committed:
Merge branch 'add_seaformer_model' of https://github.com/inderpreetsingh01/transformers into add_seaformer_model

inderpreetsingh01 commented:
Hi @alaradirik, thanks for your response, removing `num_labels` from config has resolved that testcase, can you please help with this test case as well 
`SeaformerModelTest::test_initialization - AssertionError: -6.169999778649071e-06 not found in [0.0, 1.0]`
I have normally initialized the parameters so negative values are expected.

I have looked at maskformer and segformer but not able to figure this out.

inderpreetsingh01 commented:
Hi @alaradirik thanks for reply, where should i create this test with the same name?

inderpreetsingh01 committed:
Merge branch 'huggingface:main' into add_seaformer_model

inderpreetsingh01 committed:
overwritten test_initialization

inderpreetsingh01 committed:
quality

inderpreetsingh01 committed:
Merge branch 'huggingface:main' into add_seaformer_model

inderpreetsingh01 commented:
Hi @alaradirik can you please do the final review? thanks

alaradirik reviewed (commented):
Hi @inderpreetsingh01, thanks for working on this, the PR looks great overall

There are just a few issues that need to be addressed before merging the PR. Most of these issues are minor (commented out / left-over code, non-descriptive variable names, etc.). Other than these, we favor accessing parameters used across multiple model subclasses via the config class attributes rather than passing each parameter as a separate argument.
Additionally, if a model can not pass a common modeling test due to not covering the specific approach, you'd need to overwrite it by creating a method of the same name within the `test_modeling_seaformer.py` script.

I'm adding a core maintainer for the final review, looking forward to merging this :)

amyeroberts commented:
@inderpreetsingh01 Thanks for adding this model! Ping me when the PR is ready for review (once all of @alaradirik's comments have been addressed and tests are passing).

inderpreetsingh01 commented:
@alaradirik thanks for the review, @amyeroberts sure will ping you once model is ready

inderpreetsingh01 committed:
Update docs/source/en/_toctree.yml

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update docs/source/en/model_doc/seaformer.mdx

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/activations.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/activations.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/activations.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/models/seaformer/__init__.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/models/seaformer/__init__.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/models/seaformer/__init__.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/models/seaformer/modeling_seaformer.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update src/transformers/models/seaformer/__init__.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update utils/check_repo.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>

inderpreetsingh01 committed:
Update utils/check_repo.py

Co-authored-by: Alara Dirik <8944735+alaradirik@users.noreply.github.com>
